universe/ShipDesign.cpp

```diff
@@ -19,6 +19,8 @@
 #include <cfloat>
 #include <boost/filesystem/fstream.hpp>
 
+extern FO_COMMON_API const int INVALID_DESIGN_ID = -1;
```

Comment:
I don't know if it is necessary or not.
BUT... I copied the ALL_EMPIRES assignment, which does have it. (Universe.cpp, line 73)
I figured it would be safer to copy what seems to working, rather than try to figure it out myself.